release_bot/init_repo.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+This module initializes a repo to be used by ReleaseBot
+"""
+import os
+import yaml
+
+from release_bot.utils import run_command_get_output
+
+TEMPLATE_STRING = """{{#general_title}}
+# {{{title}}}
+
+{{/general_title}}
+{{#versions}}
+
+{{#sections}}
+### {{{label}}}
+
+{{#commits}}
+* {{{subject}}} [{{{author}}}]
+{{#body}}
+
+{{{body_indented}}}
+{{/body}}
+
+{{/commits}}
+{{/sections}}
+
+{{/versions}}"""
+
+GITCHANGELOG_RC_STRING = """
+output_engine =  mustache("markdown.tpl")
+"""
+
+class Init:
+    """
+    Creates all of the required configuration script required for the ReleaseBot
+    """
+    def __init__(self):
+        self.conf = {}
+        self.release_conf = {}
+        self.create_conf()
+        self.append_to_gitignore()
+        if self.conf['gitchangelog']:
+            self.create_template()
+            self.create_gitchangelog_rc()
```

Comment:
You should call these in a separate method.
The `__init__()` should contain only the necessary initialization, nothing else.

```python
    def __init__(self):
        self.conf = {}
        self.release_conf = {}

   def hardly_working_on_it():
        self.create_conf()
        self.append_to_gitignore()
        if self.conf['gitchangelog']:
            self.create_template()
            self.create_gitchangelog_rc()

... releasebot.py ...
def main():
     if configuration.init:
        init = Init()
        init.hardly_working_on_it()
```